src/python/pants/engine/internals/scheduler_test.py

```diff
@@ -144,7 +149,7 @@ def num_wheels(self) -> int:
 
 
 @rule
-def car_num_wheels(car: Car) -> int:
+def car_num_wheels(car: Car, _: Fuel) -> int:
```

Comment:
What's the difference between this new syntax, and having a Fuel object be passed into the construction of the union member(s)?

Reply:
That would be equivalent from a "number of `@rule` instances perspective. But from a practical perspective, adjusting every type everywhere to pass a type (and then actually passing it) is precisely the kind of thing the engine was designed to avoid.

Reply:
So instead of adjusting types and callsites we adjust rules declarations? I can dig it.